packages/components/src/modifiers/hds-clipboard.ts

```diff
@@ -151,7 +151,11 @@ export const copyToClipboard = async (
 ): Promise<boolean> => {
   let textToCopy: string = '';
 
-  if (text) {
+  if (text === '') {
+    textToCopy = '';
+  } else if (text === 0) {
+    textToCopy = '0';
+  } else if (text) {
```

Comment:
[suggestion] what if we just check that the value is defined here? (potentially we could do the same also for `else if (target !== undefined)` below. 

My reasoning is: it seems more "correct" in relation to the logic in this function (the actual check about the type is done anyway inside the `getTextToCopy()` function, here is just a logical switch between the two arguments `@text` or `@target` to decide which function to invoke

```suggestion
  if (text !== undefined) {
```

Reply:
In this way you can then remove the `if (text)` check inside the `getTextToCopy()` function and the `if (textToCopy)` check inside the `writeTextToClipboard()` function

As point in favour of this: the updated logic would reflect the typing definition of the functions, that are not expecting undefined values

Reply:
I made the changes & pushed up. I also had to remove a unit test for passing an undefined value to `writeTextToClipboard`.

Reply:
@KristinLBradley check the "Copy a Bigint number" in the `Copy::Snippet`

[You may also consider adding the two missing cases, empty string and zero, directly to the first "Content" section]

Reply:
Sorry, I missed that. I removed that section and added to the Content section examples instead.

Reply:
https://hds-showcase-git-hds-4447-copybutton-empty-string-hashicorp.vercel.app/components/copy/snippet#content